rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -367,16 +368,19 @@ protected void renewLeaseForClient(Scheduler scheduler) throws NotConnectedRestE
 
             boolean onlyUserJobs = (myJobs && user != null && user.trim().length() > 0);
 
-            List<JobInfo> jobsInfo = s.getJobs(index, range,
-                    new JobFilterCriteria(onlyUserJobs, pending, running, finished), DEFAULT_JOB_SORT_PARAMS);
-            List<UserJobData> jobs = new ArrayList<>(jobsInfo.size());
+            Page<JobInfo> page = s.getJobs(index, range, new JobFilterCriteria(onlyUserJobs, pending, running, finished), DEFAULT_JOB_SORT_PARAMS);
+            List<JobInfo> jobsInfo = page.getList();
+            ArrayList<UserJobData> jobs = new ArrayList<>(jobsInfo.size());
             for (JobInfo jobInfo : jobsInfo) {
                 jobs.add(new UserJobData(mapper.map(jobInfo, JobInfoData.class)));
             }
 
-            HashMap<Long, List<UserJobData>> map = new HashMap<>(1);
+            HashMap<Long, ArrayList<UserJobData>> map = new HashMap<Long, ArrayList<UserJobData>>(1);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The type of the "map" object should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)